src/test/java/com/commercetools/pspadapter/notification/processor/impl/PaymentSalePendingProcessorTest.java

```diff
@@ -0,0 +1,84 @@
+package com.commercetools.pspadapter.notification.processor.impl;
+
+import com.commercetools.payment.constants.paypalPlus.NotificationEventType;
+import com.commercetools.pspadapter.facade.CtpFacade;
+import com.commercetools.service.ctp.CartService;
+import com.commercetools.service.ctp.OrderService;
+import com.commercetools.service.ctp.PaymentService;
+import com.commercetools.service.ctp.impl.PaymentServiceImpl;
+import com.google.gson.GsonBuilder;
+import com.paypal.api.payments.Event;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.Transaction;
+import io.sphere.sdk.payments.TransactionState;
+import io.sphere.sdk.payments.TransactionType;
+import io.sphere.sdk.payments.commands.updateactions.ChangeTransactionState;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.commercetools.testUtil.CompletionStageUtil.executeBlocking;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
```

Comment:
If you use `@MockitoJUnitRunner` you could avoid manual explicit mocking (like `mock(Payment.class)`) and use annotation instead: `@Mock Payment payment;` (and all other `mock(...)` entries). 

If u don't use `@Mock` or `@Spy` annotations, `MockitoJUnitRunner` is redundant and default JUnit runner is sufficient.

But, of course, if you have only one test case - there is no much difference.

Reply:
I tried to use annotation, but I'm struggling with this one:

```java
    @Mock
    SphereClient sphereClient;

    @Spy
    PaymentService paymentService = new PaymentServiceImpl(sphereClient);
```
How do I tell that Mockito should inject the mock object into the spy? Now I got NPE.

Anyway, even if I don't use the annotations, I think it's good for documentation purposes to have `@MockitoJUnitRunner`, so you know immediately that the class uses Mockito.

Reply:
Looks like it's (not possible like that)[https://stackoverflow.com/questions/43926636/mockito-inject-mock-into-spy-object].

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/69/commits/e2f22e28c25aafbc13c1333ddd96969db41e083a